### PR TITLE
Update log.py

### PR DESCRIPTION
**Pull request by suresttexas00**
Fix log rotation by altering how the log files are written and setting up a buffer for when the file is not available for writing.

### CODE REVIEW

File: src/log.py [hunk 1]
```diff
@@ -18,16 +19,13 @@ def rotateLogs(self):
...
-			self.file.close()
-			time.sleep(1) # Possibly fix Windows 'in-use' error. Ugh.
 		if os.path.exists("logs/wrapper/current.log"):
 			with open("logs/wrapper/current.log", "r") as f:
 				logData = f.read()
-			with gzip.open("logs/wrapper/%s.log.gz" % time.strftime("%Y-%m-%d_%H-%M-%S"), "w") as f:
-				f.write(logData)
-			os.remove("logs/wrapper/current.log")
-		self.file = open("logs/wrapper/current.log", "a")
+		os.remove("logs/wrapper/current.log")
```
Comment (benbaptist) [inline src/log.py]: You put os.remove outside of the if os.path.exists - if current.log doesn't exist, this will fail.

**Comment by suresttexas00** [comment]
That was actually intentional because I suspect that os.exists() puts a file lock on the file in Windows.  The file would normally exist anyways except under _very unusual_ circumstances that I cannot conceive of.  If you wanted to try - except it (the os.remove) perhaps?

File: src/log.py (hunk 1 above)
Comment (suresttexas00) [inline src/log.py]: Also, the sleep, whether (1) or (60000), does nothing (I tried!).  It is not a timing issue. Something about the way you wrote this maintains a file lock, even if you think it closes.  I don't quite understand what the self.file definition does for you, but (on windows) it seems to keep the file locked.  This code works, although I can understand the desirability of adding some error checking to ensure it doesn't attempt to delete a non existent file...

**Comment by benbaptist** [comment]
This should be filed as a bug report to Microsoft. How the hell have they made such a god awful system? They should have experience... after bloody 20+ years, they should know what they're doing, but obviously they do not. 

I don't get why os.path.exists would ever keep a lock - it should get rid of the lock since it's done. Why the hell must Windows be a terrible platform?

**Comment by suresttexas00** [comment]
Can we ease up on the windows vitriol?  I appreciate Linux too, but I am not running an OS inside and OS...  As far as the os.path.exists... as I understand, the method works by "opening" the file (resulting in the lock).  If it opens, "it exists", if it cannot open, it does "not exist".  maybe I am just thinking about while loops, but I thought that putting this inside the if block was not a good idea, especially as we pretty much __KNOW__ the file will exist.  If it does not exists ,we should probably be exiting the method anyway... since there is no file to archive.

However, I suspect the problem is deeper than this.  Remember my issue with how getStorage objects on the disk would fluctuate?  The item on the disk would randomly "be" some prior instance of the object, which would change every so often, as your timer wrote the file back.  The only time those files are accurate is immediately after /reload..  Now, I don't know a lot about python and threads and such, but it just seems that maybe something similar is happening here because, unless the wrapper gets shut off, when you use the self.file thing, I really suspect that some stray thread still thinks the dang thing is open (hence the file lock).

**Comment by suresttexas00** [comment]
How about this instead?

```
		self.file = None
		self.rotateLogs()
		self.buffer = ""
+		self.useBuffer = False
```

```
...
		if not os.path.exists("logs/wrapper/current.log"):
			return
		self.useBuffer = True
		if self.file:
			self.file.close()
		self.file = open("logs/wrapper/current.tmp", "w")
		with open("logs/wrapper/current.log", "r") as f:
			logData = f.read()
			f.close()
		with gzip.open("logs/wrapper/%s.log.gz" % time.strftime("%Y-%m-%d_%H-%M-%S"), "w") as f:
			f.write(logData)
		os.remove("logs/wrapper/current.log")
		self.file = open("logs/wrapper/current.log", "w")
		self.useBuffer = True
```

```
	def write(self, payload):
		try:
			if self.file and self.useBuffer is True:
				if len(self.buffer) > 0:
					self.file.write(self.buffer)
					self.buffer = ""
```

I think this _might_ work...

**Pull request closed without merging**